components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1813,9 +1814,11 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             workspaceId,
             params: {
                 ...censor(params, "config"),
-                config: {
-                    vscode: params?.config?.vscode,
-                },
+                ...(params.config?.vscode ? {
```

Comment:
Ah, I think checking `param.config` was more correct than checking `params.config.vscode`, but oh well.

We've done enough hair-splitting! 😇